pagopa/io-app
# chore: [IOPID-2405] Delete isNewPaymentSectionEnabledSelector and isSettingsVisibleAndHideProfileSelector

## Short description
delete isNewPaymentSectionEnabledSelector and isSettingsVisibleAndHideProfileSelector and all references and related logic in app

## List of changes proposed in this pull request
- delete isNewPaymentSectionEnabledSelector and all references
- delete isSettingsVisibleAndHideProfileSelector
  - delete useContentWithFF (custom hook used to assign "Profile" or "Settings" value into a string using `PageName` as placeholder) and all references and change values in the translation files 
  - delete all legacy copy from code and from the translation files
  - delete all reference about `ROUTES.PROFILE_MAIN`
- Update test and snapshots

> [!Note]
> The **German texts** were modified by inserting the value that was managed before using `useContentWithFF`. So compared to what is in production now there are no regressions

## Demo

| Android - local env (first onboarding) | Android - prod env (user logged) | iOS - local env (first onboarding) | iOS - prod env (user logged) |
| - | - | - | - |
| <video src="https://github.com/user-attachments/assets/f3d9e3ce-7d39-49ab-8225-2c9b46475a77"/> | <video src="https://github.com/user-attachments/assets/a396930c-9ca7-44d7-aad2-404ce140d9b6"/> | <video src="https://github.com/user-attachments/assets/1c097bd4-e410-4d37-bd41-c455be7cfe46"/> | <video src="https://github.com/user-attachments/assets/f8bdd3e2-66f3-42d5-b5de-c64269387c67"/> |

## How to test
- Run the application (use local and prod environment) and check
  - if the build works on Android and iOS
  - check that all screens work as expected
- run `yarn tsc -noemit` and check if everything works
- check if all tests works
